.changeset/every-loops-help.md

```diff
@@ -0,0 +1,41 @@
+---
+'@seek/logger': major
+---
+
+Enforce stricter typing for logger methods
+
+This is a **breaking change** to the types, improving type safety by enforcing stricter parameter typing on all logger methods.
+
+Existing code that passes metadata after the message string will need to be updated. This pattern was previously silently ignoring the metadata, but now triggers a type error to prevent data loss.
+
+**Before (no longer works):**
+
+```ts
+logger.error('my message', { error, metadata });
+```
+
+In this pattern, any metadata passed after the message string is not captured by the logger.
+
+**After (correct usage):**
+
+```ts
+logger.error({ error, metadata }, 'my message');
+```
+
+This ensures all metadata is properly captured and logged.
+
+The change also enforces type checking for placeholder arguments:
+
+**Before (no longer works):**
+
+```ts
+logger.error('my message with string: %s and number: %d', 99, 'string');
+```
+
+**After (correct usage):**
+
+```ts
+logger.error('my message with string: %s and number: %d', 'string', 99);
+```
+
+This prevents type mismatches between placeholders and their corresponding arguments.
```

Comment:
I'd rather us not support/broadcast this option if it's not aligned with intended usage (structured JSON logging)

Reply:
I mean it all ends up as a JSON log in the end right?

Reply:
Or would you rather I add some metadata to the log?

eg. `logger.error( { err }, 'etc')`

Reply:
Cool, I'm dropping it completely given the low usage across our repos anyway